### PR TITLE
Add ilike pushdown function for tiflash

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/13447
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by dveeden — CHANGES_REQUESTED** [review]
I don't see an `Ilike` (or `ILIKE`) on https://docs.pingcap.com/tidb/dev/operators

It also doesn't seem to be present in the last nightly build:
```
sql> SELECT "FOO" LIKE "FOO";
+------------------+
| "FOO" LIKE "FOO" |
+------------------+
|                1 |
+------------------+
1 row in set (0.0015 sec)

sql> SELECT "FOO" RLIKE "FOO";
+-------------------+
| "FOO" RLIKE "FOO" |
+-------------------+
|                 1 |
+-------------------+
1 row in set (0.0012 sec)

sql> SELECT "FOO" ILIKE "FOO";
ERROR: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 24 near ""FOO"" 

sql> SELECT tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.7.0-alpha
Edition: Community
Git Commit Hash: ee7c089e55e6d95425c745c8a9073632aa5e760e
Git Branch: heads/refs/tags/v6.7.0-alpha
UTC Build Time: 2023-03-15 14:34:49
GoVersion: go1.20.2
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv
1 row in set (0.0011 sec)
```

Could you fill in more of the pull request template with background on this and references?

**Comment by xzhangxian1008** [comment]
> I don't see an `Ilike` (or `ILIKE`) on https://docs.pingcap.com/tidb/dev/operators
> 
> It also doesn't seem to be present in the last nightly build:
> 
> ```
> sql> SELECT "FOO" LIKE "FOO";
> +------------------+
> | "FOO" LIKE "FOO" |
> +------------------+
> |                1 |
> +------------------+
> 1 row in set (0.0015 sec)
> 
> sql> SELECT "FOO" RLIKE "FOO";
> +-------------------+
> | "FOO" RLIKE "FOO" |
> +-------------------+
> |                 1 |
> +-------------------+
> 1 row in set (0.0012 sec)
> 
> sql> SELECT "FOO" ILIKE "FOO";
> ERROR: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 24 near ""FOO"" 
> 
> sql> SELECT tidb_version()\G
> *************************** 1. row ***************************
> tidb_version(): Release Version: v6.7.0-alpha
> Edition: Community
> Git Commit Hash: ee7c089e55e6d95425c745c8a9073632aa5e760e
> Git Branch: heads/refs/tags/v6.7.0-alpha
> UTC Build Time: 2023-03-15 14:34:49
> GoVersion: go1.20.2
> Race Enabled: false
> TiKV Min Version: 6.2.0-alpha
> Check Table Before Drop: false
> Store: tikv
> 1 row in set (0.0011 sec)
> ```
> 
> Could you fill in more of the pull request template with background on this and references?

ilike function was just merged into tidb. [pr](https://github.com/pingcap/tidb/pull/40942)

File: tiflash/tiflash-supported-pushdown-calculations.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ TiFlash supports the following push-down expressions:
 | Expression Type | Operations |
 | :-------------- | :------------------------------------- |
 | [Numeric functions and operators](/functions-and-operators/numeric-functions-and-operators.md) | `+`, `-`, `/`, `*`, `%`, `>=`, `<=`, `=`, `!=`, `<`, `>`, `ROUND()`, `ABS()`, `FLOOR(int)`, `CEIL(int)`, `CEILING(int)`, `SQRT()`, `LOG()`, `LOG2()`, `LOG10()`, `LN()`, `EXP()`, `POW()`, `SIGN()`, `RADIANS()`, `DEGREES()`, `CONV()`, `CRC32()`, `GREATEST(int/real)`, `LEAST(int/real)` |
-| [Logical functions](/functions-and-operators/control-flow-functions.md) and [operators](/functions-and-operators/operators.md) | `AND`, `OR`, `NOT`, `CASE WHEN`, `IF()`, `IFNULL()`, `ISNULL()`, `IN`, `LIKE`, `COALESCE`, `IS` |
+| [Logical functions](/functions-and-operators/control-flow-functions.md) and [operators](/functions-and-operators/operators.md) | `AND`, `OR`, `NOT`, `CASE WHEN`, `IF()`, `IFNULL()`, `ISNULL()`, `IN`, `LIKE`, `Ilike`, `COALESCE`, `IS` |
```
Comment (dveeden) [inline tiflash/tiflash-supported-pushdown-calculations.md:39]:
```suggestion
| [Logical functions](/functions-and-operators/control-flow-functions.md) and [operators](/functions-and-operators/operators.md) | `AND`, `OR`, `NOT`, `CASE WHEN`, `IF()`, `IFNULL()`, `ISNULL()`, `IN`, `LIKE`, `ILIKE`, `COALESCE`, `IS` |
```

**Review by dveeden — CHANGES_REQUESTED** [review]
This also needs to be added to `functions-and-operators/operators.md`.

Make sure to add a `## MySQL compatibility` section as `ILIKE` doesn't exist in MySQL and is also not in the SQL standards.

File: tiflash/tiflash-supported-pushdown-calculations.md (hunk 1 above)
Comment (joey-yez) [inline tiflash/tiflash-supported-pushdown-calculations.md:39]:
```suggestion
| [Logical functions](/functions-and-operators/control-flow-functions.md) and [operators](/functions-and-operators/operators.md) | `AND`, `OR`, `NOT`, `CASE WHEN`, `IF()`, `IFNULL()`, `ISNULL()`, `IN`, `LIKE`, `ILIKE`, `COALESCE`, `IS` |
```

**Comment by xzhangxian1008** [comment]
> This also needs to be added to `functions-and-operators/operators.md`.
> 
> Make sure to add a `## MySQL compatibility` section as `ILIKE` doesn't exist in MySQL and is also not in the SQL standards.
> 
> You might want to link to https://dev.mysql.com/doc/refman/8.0/en/case-sensitivity.html
> 
> An example for how MySQL does this:
> 
> ```
> mysql> WITH a AS (SELECT "foo" c1 UNION ALL SELECT "FOO") SELECT c1, c1 = "foo", c1 LIKE "foo", c1 LIKE "foo" COLLATE utf8mb4_0900_as_cs FROM a;
> +-----+------------+---------------+------------------------------------------+
> | c1  | c1 = "foo" | c1 LIKE "foo" | c1 LIKE "foo" COLLATE utf8mb4_0900_as_cs |
> +-----+------------+---------------+------------------------------------------+
> | foo |          1 |             1 |                                        1 |
> | FOO |          1 |             1 |                                        0 |
> +-----+------------+---------------+------------------------------------------+
> 2 rows in set (0.00 sec)
> 
> mysql> WITH a AS (SELECT "foo" COLLATE utf8mb4_0900_as_cs c1 UNION ALL SELECT "FOO") SELECT c1, c1 = "foo", c1 LIKE "foo", c1 LIKE "foo" COLLATE utf8mb4_0900_
> +------+------------+---------------+------------------------------------------+
> | c1   | c1 = "foo" | c1 LIKE "foo" | c1 LIKE "foo" COLLATE utf8mb4_0900_as_cs |
> +------+------------+---------------+------------------------------------------+
> | foo  |          1 |             1 |                                        1 |
> | FOO  |          0 |             0 |                                        0 |
> +------+------------+---------------+------------------------------------------+
> 2 rows in set (0.01 sec)
> 
> mysql> SELECT VERSION();
> +-----------+
> | VERSION() |
> +-----------+
> | 8.0.32    |
> +-----------+
> 1 row in set (0.00 sec)
> ```
> 
> Maybe discuss with the documentation team if we want to document this as PostgreSQL compatibility? If we do, maybe link to https://www.postgresql.org/docs/current/functions-matching.html
> 
> Also looks like Snowflake has this as well: https://docs.snowflake.com/en/sql-reference/functions/ilike

okk

File: functions-and-operators/operators.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ This document describes the operators precedence, comparison functions and opera
 | [<](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#operator_less-than) | Less than operator |
 | [<=](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#operator_less-than-or-equal) | Less than or equal operator |
 | [LIKE](https://dev.mysql.com/doc/refman/5.7/en/string-comparison-functions.html#operator_like) | Simple pattern matching |
+| [ILIKE](https://www.postgresql.org/docs/current/functions-matching.html) | Simple pattern matching with case insensitive(Supported in TiDB, but not supported in MySQL) |
```
Comment (qiancai) [inline functions-and-operators/operators.md:44]:
```suggestion
| [ILIKE](https://www.postgresql.org/docs/current/functions-matching.html) | Simple pattern matching with case insensitive (Supported in TiDB, but not supported in MySQL) |
```

File: functions-and-operators/operators.md [hunk 3]
```diff
@@ -109,6 +110,7 @@ For details, see [Operator Precedence](https://dev.mysql.com/doc/refman/5.7/en/o
...
```
Comment (qiancai) [inline functions-and-operators/operators.md:113]:
```suggestion
| [ILIKE](https://www.postgresql.org/docs/current/functions-matching.html) | Simple pattern matching with case insensitive (Supported in TiDB, but not supported in MySQL) |
```

**Review by dveeden — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Pull request merged**